lms/djangoapps/course_home_api/outline/v1/views.py

```diff
@@ -158,11 +180,29 @@ def get(self, request, *args, **kwargs):
         transformers = BlockStructureTransformers()
         transformers += get_course_block_access_transformers(request.user)
         transformers += [
-            BlocksAPITransformer(None, None, depth=3),
+            BlocksAPITransformer(None, None, depth=4),
         ]
 
         course_blocks = get_course_blocks(request.user, course_usage_key, transformers, include_completion=True)
 
+        has_visited_course = False
+        try:
+            resume_block = get_key_to_last_completed_block(request.user, course.id)
+            has_visited_course = True
+        except UnavailableCompletionData:
+            resume_block = course_usage_key
```

Comment:
All of that previous logic was unnecessary because the Completion API has a method for this! 🎉